pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/DoubleMutableDictionary.java

```diff
@@ -33,40 +44,41 @@ public void index(Object rawValue) {
 
     if (rawValue instanceof String) {
       Double entry = Double.parseDouble((String) rawValue);
-      addToDictionaryBiMap(entry);
+      _dictionaryIdBiMap.put(entry);
       updateMinMax(entry);
       return;
     }
 
     if (rawValue instanceof Double) {
-      addToDictionaryBiMap(rawValue);
-      updateMinMax((Double) rawValue);
+      Double doubleValue = (Double) rawValue;
+      _dictionaryIdBiMap.put(doubleValue);
+      updateMinMax(doubleValue);
       return;
     }
 
     if (rawValue instanceof Object[]) {
       for (Object o : (Object[]) rawValue) {
         if (o instanceof String) {
           final Double doubleValue = Double.parseDouble(o.toString());
```

Comment:
repeated code blocks. Move those to internal method. We are also inconsistent with o.toString() vs (String) o for string instances. Will be good to just (String) o